packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1591,6 +1674,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (widget.selectionEnabled && pasteEnabled && widget.selectionControls?.canPaste(this) == true) {
       _clipboardStatus?.update();
     }
+
+    // If text direction has changed then our _whitespaceFormatter needs to be
+    // rerun.
+    if (_lastUsedWhitespaceFormatter?._baseDirection != _textDirection) {
+      _value = _whitespaceFormatter.format(_value);
```

Comment:
The whitespaceFormatter is added without the developer knowing, and it may introduce changes that neither the user nor the developer expects (e.g. when the input field expects a password). Maybe we should introduce an opaque layer that does this in `RenderEditable` that abstracts this away?

Reply:
Is there no way to disable it currently?

Reply:
I don't think so. I should probably remove it from here to keep the current behavior.